tests/test_encryption.py

```diff
@@ -3,53 +3,74 @@
-        "unencrypted.pdf",
+        ("unencrypted.pdf", False),
         # created by `qpdf --encrypt "" "" 40 -- unencrypted.pdf r2-empty-password.pdf`
-        "r2-empty-password.pdf",
+        ("r2-empty-password.pdf", False),
         # created by `qpdf --encrypt "" "" 128 -- unencrypted.pdf r3-empty-password.pdf`
-        "r3-empty-password.pdf",
+        ("r3-empty-password.pdf", False),
         # created by `qpdf --encrypt "asdfzxcv" "" 40 -- unencrypted.pdf r2-user-password.pdf`
-        "r2-user-password.pdf",
+        ("r2-user-password.pdf", False),
         # created by `qpdf --encrypt "asdfzxcv" "" 128 -- unencrypted.pdf r3-user-password.pdf`
-        "r3-user-password.pdf",
+        ("r3-user-password.pdf", False),
         # created by `qpdf --encrypt "asdfzxcv" "" 128 --force-V4 -- unencrypted.pdf r4-user-password.pdf`
-        "r4-user-password.pdf",
+        ("r4-user-password.pdf", False),
         # created by `qpdf --encrypt "asdfzxcv" "" 128 --use-aes=y -- unencrypted.pdf r4-aes-user-password.pdf`
-        "r4-aes-user-password.pdf",
+        ("r4-aes-user-password.pdf", True),
         # # created by `qpdf --encrypt "" "" 256 --force-R5 -- unencrypted.pdf r5-empty-password.pdf`
-        "r5-empty-password.pdf",
+        ("r5-empty-password.pdf", True),
         # # created by `qpdf --encrypt "asdfzxcv" "" 256 --force-R5 -- unencrypted.pdf r5-user-password.pdf`
-        "r5-user-password.pdf",
+        ("r5-user-password.pdf", True),
         # # created by `qpdf --encrypt "" "asdfzxcv" 256 --force-R5 -- unencrypted.pdf r5-owner-password.pdf`
-        "r5-owner-password.pdf",
+        ("r5-owner-password.pdf", True),
         # created by `qpdf --encrypt "" "" 256 -- unencrypted.pdf r6-empty-password.pdf`
-        "r6-empty-password.pdf",
+        ("r6-empty-password.pdf", True),
         # created by `qpdf --encrypt "asdfzxcv" "" 256 -- unencrypted.pdf r6-user-password.pdf`
-        "r6-user-password.pdf",
+        ("r6-user-password.pdf", True),
         # created by `qpdf --encrypt "" "asdfzxcv" 256 -- unencrypted.pdf r6-owner-password.pdf`
-        "r6-owner-password.pdf",
+        ("r6-owner-password.pdf", True),
     ],
 )
-def test_encryption(name):
+def test_encryption(name, requres_pycryptodome):
     inputfile = os.path.join(RESOURCE_ROOT, "encryption", name)
-    ipdf = PyPDF2.PdfReader(inputfile)
-    if inputfile.endswith("unencrypted.pdf"):
-        assert not ipdf.is_encrypted
+    if requres_pycryptodome and not HAS_PYCRYPTODOME:
+        with pytest.raises(DependencyError) as exc:
+            ipdf = PyPDF2.PdfReader(inputfile)
+            if inputfile.endswith("unencrypted.pdf"):
+                assert not ipdf.is_encrypted
+            else:
+                assert ipdf.is_encrypted
+                ipdf.decrypt("asdfzxcv")
+            assert len(ipdf.pages) == 1
+            dd = dict(ipdf.metadata)
+        assert exc.value.args[0] == "PyCryptodome is required for AES algorithm"
+        return
```

Comment:
If I understand the test right, this could be simplified to:
```suggestion
    if requres_pycryptodome and not HAS_PYCRYPTODOME:
        with pytest.raises(DependencyError) as exc:
            ipdf = PyPDF2.PdfReader(inputfile)
            ipdf.decrypt("asdfzxcv")
        assert exc.value.args[0] == "PyCryptodome is required for AES algorithm"
        return
```

as the exception gets thrown from the `decyrpt` function? Could further simplify this (and the else branch) by lifting the `ipdf = PyPDF2.PdfReader(inputfile)` out of the if/else altogether. Given the `return` here, I'd recommend dropping the `else` and de-indent that clause, just as a general hatred of indentation and desire to avoid it where possible.

Reply:
I can give it a try, but I doubt that your suggested simplification would work. The `unencrypted.pdf` is not encrypted, so the pycryptodome depenceny is not necessary for it.

Reply:
I think for that file's testcase, it has `requres_pycryptodome is False`, and so should fail to enter the block?

Reply:
Oh, right :+1: Thank you :pray: